### PR TITLE
C#: Correctly parse operator names in MaD

### PR DESCRIPTION
**Pull request by hvitved**
We were incorrectly using the operator name (e.g. `+`) instead of the operator function name (e.g. `op_Addition`).

### CODE REVIEW

File: csharp/ql/lib/semmle/code/csharp/dataflow/internal/ExternalFlow.qll [hunk 1]
```diff
@@ -310,10 +328,17 @@ class UnboundCallable extends Callable {
...
+  or
+  not d instanceof Operator and
+  d.hasName(name)
+}
+
 pragma[nomagic]
 private predicate callableSpecInfo(Callable c, string namespace, string type, string name) {
   c.getDeclaringType().hasQualifiedName(namespace, type) and
-  c.getName() = name
+  hasName(c, name)
```
Comment (michaelnebel) [inline csharp/ql/lib/semmle/code/csharp/dataflow/internal/ExternalFlow.qll:341]: 🙀 So was it the case, that even though the models existed they were not applied correctly to the operator?

File: csharp/ql/lib/semmle/code/csharp/dataflow/internal/ExternalFlow.qll [hunk 2]
```diff
@@ -87,27 +87,45 @@
...
+private import FlowSummaryImpl::Private::External
+private import FlowSummaryImplSpecific
 private import codeql.mad.ModelValidation as SharedModelVal
 
-/** Holds if a source model exists for the given parameters. */
-predicate sourceModel = Extensions::sourceModel/9;
+/**
+ * Holds if a source model exists for the given parameters.
+ */
+extensible predicate sourceModel(
```
[the remark this answers is not in the document]
Comment (hvitved) [reply csharp/ql/lib/semmle/code/csharp/dataflow/internal/ExternalFlow.qll:102]: That means those predicates are exposed; do we really want/need that?

File: csharp/ql/lib/semmle/code/csharp/dataflow/internal/ExternalFlow.qll (hunk 1 above)
Comment (hvitved) [reply csharp/ql/lib/semmle/code/csharp/dataflow/internal/ExternalFlow.qll:341]: Yes.

**Review by michaelnebel — APPROVED** [review]
LGTM 👍

**Pull request merged**